<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="default_robot"/>

  <group ns="$(arg robot_name)/detection">
    <!-- color detector -->
    <node name="image_republish" pkg="image_transport" type="republish" args="compressed raw">
      <remap from="in" to="/$(arg robot_name)/device/sensor0/head_camera/image_raw" />
      <remap from="out" to="image_raw" />
    </node>
    <node name="coke_detector" pkg="s4_sensor" type="coke_detect.py" >
      <remap from="input_image" to="image_raw" />
    </node>
  
    <!-- obstacle detector -->
    <rosparam command="load" file="$(find s4_sensor)/config/obstacle_detector.yaml" />
    <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
      <param name="frame_id" value="$(arg robot_name)/map" />
      <remap from="scan" to="/$(arg robot_name)/device/sensor0/front_laser/scan" />
    </node>
    <node name="obstacle_tracker" pkg="obstacle_detector" type="obstacle_tracker_node">
      <param name="frame_id" value="$(arg robot_name)/map" />
    </node>
  </group>

  <!-- merge detector -->
</launch>

